src/main/java/ui/BoardAutoCreator.java

```diff
@@ -84,8 +84,13 @@ public Menu generateBoardAutoCreateMenu() {
         return autoCreate;
     }
 
-    private void saveBoardAfterUserConfirmation(String boardName) {
-        if (isSaveBoardDialogResponsePositive(boardName)) ui.getMenuControl().saveBoardAs();
+    private void saveBoardAfterUserConfirmationIfNeeded(String boardName) {
```

Comment:
The confusing name of this method might be a call to refactor some methods or maybe add a comment explaining the `IfNeeded` and `UserConfirmation` part.